src/utils/template_handler.rs

```diff
@@ -288,6 +393,71 @@ fn load_remote_template_dir(path: &str, url: &str, force: bool) -> Status {
     Status::ok()
 }
 
+fn load_remote_gitlab_template_dir(path: &str, url: &str, force: bool) -> Status {
+    if !force && Path::new(path).exists() {
+        return Status::error(format!(
+            "Directory {} already exists...",
+            path.replace(".templates/", "")
+        ));
+    }
+
+    if !Path::new(path).exists() {
+        std::fs::create_dir(path).unwrap();
+    }
+
+    let response = rest::json_call(url);
+    if response.is_err() {
+        return Status::error(format!(
+            "Failed to get template from {}: : Request failed",
+            url
+        ));
+    }
+    let response = response.unwrap().json();
+    if response.is_err() {
+        return Status::error(format!(
+            "Failed to get template from {}: JSON parse error",
+            url
+        ));
+    }
+    let response: serde_json::Value = response.unwrap();
+    let items = response.as_array().unwrap();
+
+    for item in items {
+        if item["type"] == "tree" {
+            let st = load_remote_gitlab_template_dir(
+                format!("{}/{}", path, item["name"])
```

Comment:
Same here with the lambdas. Maybe then it make sense to add a function for this...